src/http/routers/v1/at_tools.rs

```diff
@@ -31,3 +41,69 @@ pub async fn handle_v1_tools(
         .unwrap()
     )
 }
+
+pub async fn handle_v1_tools_validate(
+    Extension(gcx): Extension<Arc<ARwLock<GlobalContext>>>,
+    body_bytes: hyper::body::Bytes,
+) -> Result<Response<Body>, ScratchError> {
+    let post = serde_json::from_slice::<ToolsPermissionCheckPost>(&body_bytes)
+        .map_err(|e| ScratchError::new(StatusCode::UNPROCESSABLE_ENTITY, format!("JSON problem: {}", e)))?;
+
+    let all_tools = tools_merged_and_filtered(gcx.clone()).await;
+
+    let mut result_messages = vec![];
+    let mut generic_tool_config = None;
+    for tool_call in &post.tool_calls {
+        let tool = match all_tools.get(&tool_call.function.name) {
+            Some(x) => x,
+            None => {
+                return Err(ScratchError::new(StatusCode::UNPROCESSABLE_ENTITY, format!("Unknown tool: {}", tool_call.function.name)))
+            }
+        };
+
+        let args = match serde_json::from_str::<HashMap<String, Value>>(&tool_call.function.arguments) {
+            Ok(args) => args,
+            Err(e) => {
+                return Err(ScratchError::new(StatusCode::UNPROCESSABLE_ENTITY, format!("JSON problem: {}", e)));
+            }
+        };
+
+        let command_to_match = {
+            let tool_locked = tool.lock().await;
+            tool_locked.command_to_match_against_confirm_deny(&args)
+        }.map_err(|e| {
+            ScratchError::new(StatusCode::UNPROCESSABLE_ENTITY, format!("Error getting tool command to match: {}", e))
```

Comment:
INTERNAL_SERVER_ERROR, not a problem with input json

Reply:
The reason this could fail without panic is due to failure parsing the command from the input json.